osvr/RenderKit/RenderManagerD3DBase.cpp

```diff
@@ -211,23 +228,19 @@ namespace renderkit {
 
         bool ret = RenderManager::PresentRenderBuffersInternal(buffers, renderInfoUsed, renderParams, normalizedCroppingViewports, flipInY);
 
-        {
-            std::set<ID3D11Texture2D*> unlockedTextures;
-            for (int i = static_cast<int>(buffers.size()) - 1; i >= 0; i--) {
-                auto colorBuffer = buffers[i].D3D11->colorBuffer;
-                if (unlockedTextures.find(colorBuffer) == unlockedTextures.end()) {
-                    unlockedTextures.insert(colorBuffer);
-                    D3D11_TEXTURE2D_DESC desc = { 0 };
-                    colorBuffer->GetDesc(&desc);
-                    if ((desc.MiscFlags & D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX) == D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX) {
-                        IDXGIKeyedMutex* mutex = nullptr;
-                        hr = colorBuffer->QueryInterface(__uuidof(IDXGIKeyedMutex), (LPVOID*)&mutex);
-                        if (!FAILED(hr) && mutex != nullptr) {
-                            hr = mutex->ReleaseSync(0); // ignore failure
-                        }
-                    }
+        // Unlock any buffers we succeeded in locking.
+        auto it = lockedTextures.begin();
+        while (it != lockedTextures.end()) {
+            D3D11_TEXTURE2D_DESC desc = { 0 };
+            (*it)->GetDesc(&desc);
+            if ((desc.MiscFlags & D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX) == D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX) {
+                IDXGIKeyedMutex* mutex = nullptr;
+                hr = (*it)->QueryInterface(__uuidof(IDXGIKeyedMutex), (LPVOID*)&mutex);
+                if (!FAILED(hr) && mutex != nullptr) {
+                    hr = mutex->ReleaseSync(0); // ignore failure
```

Comment:
Since we are no longer ignoring failure on `AcquireSync`, should we do something about a failure on `ReleaseSync`? Maybe just a throttled log message?